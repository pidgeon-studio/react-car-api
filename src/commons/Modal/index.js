import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Modal extends Component {
  getComponent(key) {
    return this.props.children.filter((c) => {
      return c.key === key;
    });
  }

  renderModal() {
    return (
      <div className="modal" tabIndex="-1" role="dialog">
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title">
                { this.props.title }

                <button type="button" className="close" onClick={this.props.close}>
                  <span aria-hidden="true">&times;</span>
                </button>
              </h5>
            </div>
            <div className="modal-body">
              { this.getComponent('body') }
            </div>
            <div className="modal-footer">
              { this.getComponent('close') }

              { this.getComponent('submit') }
            </div>
          </div>
        </div>
      </div>
    );
  }

  render() {
    return this.props.isOpen ? this.renderModal() : null;
  }
}

Modal.propTypes = {
  children: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired,
  close: PropTypes.func.isRequired
};

export default Modal;