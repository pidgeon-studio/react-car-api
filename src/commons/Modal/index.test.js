import React from 'react';
import Modal from './';
import renderer from 'react-test-renderer';

it('Modal renders correctly', () => {
  const tree = renderer
    .create(<Modal isOpen={true} title="Modal title" close={() => {}}>
      <p key="body">Body content</p>
      <button key="submit">Submit</button>
      <button key="close">Close</button>
    </Modal>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});