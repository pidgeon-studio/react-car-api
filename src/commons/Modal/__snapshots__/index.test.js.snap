// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div
  className="modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Modal title
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <p>
          Body content
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button>
          Close
        </button>
        <button>
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
